### PR TITLE
docs(Modal): added section(s) on changing the background for the modal

### PR DESCRIPTION
**Pull request by zero0Halo**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [ ] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config
- [ ] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

My code does...

### CODE REVIEW

File: docs/src/pages/components/modal.mdx [hunk 1]
```diff
@@ -188,10 +191,53 @@ And voila! Here is our `PageExample` component rendered with our `ModalAndButton
...
+uninstantiated `styled-component` to the `ModalProvider` as a prop.
+
+```jsx
+import { BaseModalBackground, ModalProvider } from '@retailmenot/anchor';
+import styled from '@xstyled/styled-components';
+
+const CustomBackground = styled(BaseModalBackground)`
+    padding: 0;
+`;
+
```
Comment (jcristol) [inline docs/src/pages/components/modal.mdx]: I love this example

**Review by jcristol — APPROVED** [review]

File: docs/src/pages/components/modal.mdx [hunk 2]
```diff
@@ -188,10 +191,53 @@ And voila! Here is our `PageExample` component rendered with our `ModalAndButton
...
+If there is a need to have a custom background for the modal, it can be accomplished by sending an
+uninstantiated `styled-component` to the `ModalProvider` as a prop.
+
+```jsx
+import { BaseModalBackground, ModalProvider } from '@retailmenot/anchor';
+import styled from '@xstyled/styled-components';
+
+const CustomBackground = styled(BaseModalBackground)`
+    padding: 0;
```
Comment (brandonlilly) [inline docs/src/pages/components/modal.mdx:204]: In this particular case, we do actually have a mechanism via props to accomplish this:

```jsx
<Modal backgroundProps={{ padding: '0', opacity: 1 }}>
  ...
</Modal>
```

Though `padding` and `opacity` are the only `backgroundProps` props we currently support

**Review by brandonlilly — APPROVED** [review]

File: docs/src/pages/components/modal.mdx (hunk 2 above)
Comment (zero0Halo) [reply docs/src/pages/components/modal.mdx:204]: I'll add this info in.
